### DISCUSSION TITLE
Activation function in GCNConv layer.

### DISCUSSION THREAD

**Discussion by errhernandez** [pyg-team/pytorch_geometric#4389] [Q&A] [answered]
Just to check that I understand: in the Kipf and Welling paper (https://arxiv.org/abs/1609.02907) describing the GCNConv, I understand that there is an activation function applied, like here (the sigma)

![image](https://user-images.githubusercontent.com/9249275/161093776-64601715-fca7-4fdd-b1a9-50ce01d994b0.png)

But in the PyG implementation the activation function is missing; I presume this is because one can just apply it to the output of the GCNConv layer, right?

Sorry if this is a stupid question. I'm a slow learner ;-)

**Comment by rusty1s** [comment, marked as answer]
Yes, activation functions are not part of our `nn.conv` layers. This has mostly two reasons:
* It makes it easier to try out different non-linearities
* It aligns well with operators from PyTorch, e.g., [`Conv2d`](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html?highlight=conv#torch.nn.Conv2d), which also do not apply a non-linearity
